agent/src/main.rs

```diff
@@ -117,6 +140,63 @@ impl ZFSDataset {
             bail!("zfs create failed! out:{} err:{}", out, err);
         }
 
+        // If requested, set a quota and reservation here.
+        if let Some(reservation) = reservation {
```

Comment:
I'd suggest we might want something like this:

```rust
        let mut cmd = std::process::Command::new("zfs")
            .arg("create");

    if let Some(reservation) = reservation {
        cmd = cmd.arg("-o").arg(format!("reservation={}", reservation));
    }

        cmd = cmd
            .arg(&dataset)
            .output()?;

```

i.e. I think you can create the dataset with the properties preset.

Reply:
Yeah, I like it.  Fewer lines of code and easier cleanup.